### ISSUE TITLE
dotnet-watch seems to hide Console.ReadLine input

### ISSUE THREAD

**Issue by mellinoe** [aspnet/dotnet-watch#128]
Consider this program:

```CSharp
using System; 
 
namespace ConsoleApplication 
{ 
    public class Program 
    { 
        public static void Main(string[] args) 
        { 
            Console.WriteLine("Hello World!"); 
            Console.WriteLine("Enter something."); 
            string input =  Console.ReadLine(); 
            Console.WriteLine($"You said {input}."); 
        } 
    } 
}
```

If I run this program using `dotnet watch run`, my input for the `Console.ReadLine()` command is not visible. Once I hit enter, the program continues as normal and my input AND output are displayed.

```
Hello World!
Enter something.
I can't see what I'm typing.
You said I can't see what I'm typing..
[DotNetWatcher] info: dotnet exit code: 0
[DotNetWatcher] info: Waiting for a file to change before restarting dotnet...
```

**Comment by muratg** [comment]
@victorhurdugaci is this by design somehow?

**Comment by victorhurdugaci** [comment]
I've tried to reproduce this but it seems to work fine. I can see the input as I type it. Is this happening on a particular OS or shell?

**Comment by Sineaggi** [comment]
I can repro this on osx. Happens in Terminal and iTerm, sh and zsh.

**Comment by natemcmaster** [comment]
This is due to a bug in how dotnet-cli forwards stdout/stdin streams. This was fixed in https://github.com/dotnet/cli/pull/3808, which should be in the next release of dotnet-cli. I tested the problem is resolved in one of the latest nightlies, 1.0.0-preview3-003546.

Closing as the issue is external to dotnet-watch.
